### PR TITLE
fix(gomod): support remote option in go_repository

### LINKED ISSUE #2832: Bazel: go_repository remote not used
**What Renovate type are you using?**
Github App

**Describe the bug**
Renovate is fetching the `importpath` repository rather than the explicit `remote` repository in the go_repository rule

**Did you see anything helpful in debug logs?**
Yes, debug logs validate my suspicion https://renovatebot.com/dashboard#github/prysmaticlabs/prysm/28753154

**To Reproduce**
See this PR: https://github.com/prysmaticlabs/prysm/pull/757

See this style of dependency

```
go_repository(
    name = "com_github_ethereum_go_ethereum",
    importpath = "github.com/ethereum/go-ethereum",
    remote = "https://github.com/prysmaticlabs/bazel-go-ethereum",
    vcs = "git",
    commit = "f4b3f83362a4cf2928e57914af040aea76c8a7d6",
)
```

**Expected behavior**

Renovate reads github.com/prysmaticlabs/bazel-go-ethereum to fetch the latest digest.

### PR DESCRIPTION
**Pull request by Nyarum**
Closes #2832

### CODE REVIEW

File: lib/manager/bazel/extract.js [hunk 1]
```diff
@@ -101,8 +101,10 @@ function extractPackageFile(content) {
       dep.depName = depName;
       dep.currentValue = currentValue || commit.substr(0, 7);
       dep.purl = 'pkg:go/' + importpath;
-      if (remote) {
+      if (remote && !remote.match(/\.git/)) {
```
Comment (rarkins) [inline lib/manager/bazel/extract.js]: The check must be more strict. It should only match URLS like https://github.com/foo/bar and nothing else

**Comment by rarkins** [comment]
@Nyarum 	I added an additional test to test for non-github remotes. The check needs to be stricter and limit only to github.com

**Comment by Nyarum** [comment]
@rarkins added fixes for strict only `github` and some fix for case with special symbol in path

**Comment by rarkins** [comment]
@Nyarum 	I thought of another edge case and added it to the tests. We need it to simply be https://github.com/owner/repo or nothing.

**Comment by Nyarum** [comment]
@rarkins covered

**Pull request merged**